js/engine/src/index.d.ts

```diff
@@ -12,7 +12,8 @@ export interface Engine {
   execute: (
     language: string,
     rulesFilename: string,
-    targetFilename: string
+    root: string,
```

Comment:
Would be nice to document a bit better those interface files (they are like .mli, and we usually try hard
to have good documentation in it, to better understand the APIs).

Reply:
For example it's not obvious why we take a list of files and a root now instead of a single file.

Reply:
Yea I'll add some more comments